CRM/Utils/Check.php

```diff
@@ -209,7 +212,7 @@ public static function checkAll($max = FALSE) {
 
     foreach ($messages as $key => $message) {
       $hush = self::checkHushSnooze($message);
-      $message->setVisible(!$hush);
+      $messages[$key]->setVisible(!$hush);
```

Comment:
@agh1, this is an important difference between how PHP handles objects and associative-arrays, and you'll find a lot of other code which relies on PHP's object support. Notably, objects in PHP feel like objects in most other languages -- you don't need to do anything special with references or pointers. Suggestion: if you see arrow notation ("->"), then assume that it applies directly to the original object/record/data.

I'm neutral on the change itself. Both the original and the modified are equally clear to me.

@colemanw, -1 on an explicit reference (`foreach (... as &$message)`). Been burned a few times by extraneous references, so think we should only use them if necessary.